x-pack/plugin/watcher/src/main/java/org/elasticsearch/xpack/watcher/execution/CurrentExecutions.java

```diff
@@ -63,9 +69,11 @@ public void remove(String id) {
      * Calling this method makes the class stop accepting new executions and throws and exception instead.
      * In addition it waits for a certain amount of time for current executions to finish before returning
      *
-     * @param maxStopTimeout    The maximum wait time to wait to current executions to finish
+     * @param maxStopTimeout The maximum wait time to wait to current executions to finish
+     * @param stoppedListener The listener that will set Watcher state to: {@link WatcherState#STOPPED}, may be {@code null} assuming the
+     *                       {@link WatcherState#STOPPED} is set elsewhere or not needed to be set.
      */
-    void sealAndAwaitEmpty(TimeValue maxStopTimeout) {
+    void sealAndAwaitEmpty(TimeValue maxStopTimeout, @Nullable Consumer<Void> stoppedListener) {
```

Comment:
you could pass a runnable here, which is still a functional interface but ignores the return value if it is void anyway? Same applies for the consumers below, won't repeat.

Reply:
done in https://github.com/elastic/elasticsearch/pull/43939/commits/39b0336b38555d3a2590c063068cd595ee327210